components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/model/AuthenticatingUser.java

```diff
@@ -76,5 +88,19 @@ public void setSub(String sub) {
     public String getSub() {
         return sub;
     }
+
+    private boolean isMultiValuedAttribute(String claimKey, String claimValue) {
+
+        // Address claim contains multi attribute separator but its not a multi valued attribute.
+        if (claimKey.equals(ADDRESS)) {
+            return false;
+        }
+        // To format the groups claim to always return as an array, we should consider single group as
+        // multi value attribute.
+        if (claimKey.equals(GROUPS)) {
+            return true;
+        }
```

Comment:
Also, we are handling groups separately right ? So this logic should not be there